src/pages/inbox/BoxesPage.tsx

```diff
@@ -17,18 +18,61 @@ import {
   mail,
   personCircleOutline,
 } from "ionicons/icons";
-import { useAppDispatch } from "../../store";
-import { getInboxCounts } from "../../features/inbox/inboxSlice";
-import { MouseEvent, useContext } from "react";
+
+import { isPostReply } from "./RepliesPage";
+import { jwtSelector } from "../../features/auth/authSlice";
 import { PageContext } from "../../features/auth/PageContext";
+import AppContent from "../../features/shared/AppContent";
+import { InsetIonItem, SettingLabel } from "../../features/user/Profile";
+import { getInboxCounts } from "../../features/inbox/inboxSlice";
+import useClient from "../../helpers/useClient";
+import { useAppDispatch, useAppSelector } from "../../store";
+
+const UnreadBubble: FC<{ count: number }> = ({ count }) =>
+  count ? (
+    <IonBadge color="danger" slot="end">
+      {count}
+    </IonBadge>
+  ) : null;
 
 export default function BoxesPage() {
+  const client = useClient();
+  const jwt = useAppSelector(jwtSelector);
   const dispatch = useAppDispatch();
+  const {
+    mentions: unreadMentions,
+    messages: unreadMessages,
+    replies: unreadReplies,
+  } = useAppSelector((state) => state.inbox.counts);
+  const [unreadPostReplies, setUnreadPostReplies] = useState(0);
+  const [unreadCountsResolved, setUnreadCountsResolved] = useState(false);
+
+  const getUnreadPostReplyCount = async (): Promise<void> => {
+    if (!jwt) return;
+
+    const allUnreads = await client.getReplies({
+      limit: 50,
```

Comment:
What if unreads is more than 50? I believe this will cause potentially inaccurate counts? (I've had >50 unreads a few times already).

We could potentially show >X?

Also, its unfortunate that we have to fetch 50 items just to display some counts :(

Reply:
Agreed on all counts (no pun intended 😭). It _is_ unfortunate we have to iterate over all unreads just to count the post replies, but I suppose it's the only option to get a valid 'post replies' count until the BE implements this count. That said, I noticed the app is fairly consistent with caching calls fwiw—this is also the reason I left the `limit` at 50, as it's unusual for the "average" user to have unreads higher than 50. There's nothing from keeping us upping this number of course; e.g. in the event a user has 500 unreads, iterating over a simple object with 500 entries without any additional processing is still peanuts (in terms of processing) for any device running this PWA.

Again, your call :)